service/src/integrationTest/kotlin/fi/espoo/evaka/pis/controller/PersonControllerIntegrationTest.kt

```diff
@@ -137,6 +141,54 @@ class PersonControllerIntegrationTest : AbstractIntegrationTest() {
         assertEquals(person.id, response.first().id)
     }
 
+    @Test
+    fun `Guardian blocklist prevents dependants from being added from VTJ data`() {
+        val guardianId = db.transaction { tx ->
+            tx.insertTestPerson(
+                DevPerson(
+                    lastName = "Karhula",
+                    firstName = "Johannes Olavi Antero Tapio",
+                    ssn = "070644-937X"
+                )
+            )
+        }
+
+        val dependants = controller.getPersonDependants(Database(jdbi), admin, guardianId)
+        assertEquals(3, dependants.size)
+
+        val blockedDependant = dependants.find { it.socialSecurityNumber == "070714A9126" }!!
+        db.transaction { tx ->
+            tx.createUpdate("DELETE FROM guardian WHERE child_id = :id").bind("id", blockedDependant.id).execute()
+            tx.insertTestGuardianBlocklistEntry(DevGuardianBlocklistEntry(guardianId, blockedDependant.id))
+        }
+
+        assertEquals(2, controller.getPersonDependants(Database(jdbi), admin, guardianId).size)
```

Comment:
I don't think dependants is queried again from VTJ here since `vtjGuardiansQueried` should not be null anymore. You should reset it or change the name of the test if it's intentional.